webpack/webpack5-module-minifier-plugin/src/AsyncImportCompressionPlugin.ts

```diff
@@ -0,0 +1,314 @@
+                    asyncImportMap.get(module);
+                  if (!localAsyncImports) {
+                    asyncImportMap.set(module, (localAsyncImports = new Map()));
+                  }
+
+                  const chunkGroups: ChunkGroup[] = targetModule.blocks.map(chunkGraph.getBlockChunkGroup);
+                  const chunkIds: Set<number | string | null> = new Set();
+
+                  for (const chunkGroup of chunkGroups) {
+                    for (const chunk of chunkGroup.chunks) {
+                      chunkIds.add(chunk.id);
+                    }
+                  }
+
+                  const idString: string = Array.from(chunkIds).join(';');
+
+                  let meta: IAsyncImportMetadata | undefined = asyncImportGroups.get(idString);
+                  if (!meta) {
+                    asyncImportGroups.set(
+                      idString,
+                      (meta = { chunkCount: chunkIds.size, chunkIds: [], count: 0, index: -1 })
+                    );
+                  }
+
+                  meta.count++;
+
+                  const stringKey: string = `${targetModule.id}`.replace(/[^A-Za-z0-9_$]/g, '_');
+
+                  const key: string = `${ASYNC_IMPORT_PREFIX}${stringKey}`;
+                  localAsyncImports.set(key, {
+                    meta,
+                    module: targetModule
+                  });
+                }
+              }
+            });
+          }
+        }
+
+        const rankedImports: [string, IAsyncImportMetadata][] = [...asyncImportGroups]
+          .filter((x) => x[1].count > 1)
+          .sort((x, y) => {
+            let diff: number = y[1].count - x[1].count;
+            if (!diff) {
+              diff = y[1].chunkCount - x[1].chunkCount;
+            }
+
+            if (!diff) {
+              diff = x[0] > y[0] ? 1 : x[0] < y[0] ? -1 : 0;
+            }
+
+            return diff;
+          });
+
+        for (let i: number = 0, len: number = rankedImports.length; i < len; i++) {
+          rankedImports[i][1].index = i;
+        }
+
+        rankedImportGroups = rankedImports.map((x) => x[1]);
+
+        const { dependencyTemplates } = compilation;
+
+        const defaultImplementation: IImportDependencyTemplate | undefined =
+          dependencyTemplates.get(ImportDependency);
+
+        if (!defaultImplementation) {
+          compilation.errors.push(new WebpackError(`Could not find ImportDependencyTemplate`));
+        }
+
+        const customTemplate: IImportDependencyTemplate = {
+          apply(
+            dep: WebpackImportDependency,
+            source: sources.ReplaceSource,
+            dependencyTemplateContext: DependencyTemplateContext
+          ): void {
+            const targetModule: Module = moduleGraph.getModule(dep);
+
+            if (targetModule) {
+              const moduleId: string | number = chunkGraph.getModuleId(targetModule);
```

Comment:
One of the lines in here should be:
```ts
dependencyTemplateContext.runtimeRequirements.add(COMPRESSED_ASYNC_IMPORT_FUNCTION);
```
Where `const COMPRESSED_ASYNC_IMPORT_FUNCTION = '__webpack_require__.E'`